cmd/docker/docker.go

```diff
@@ -419,10 +419,11 @@ func areSubcommandsSupported(cmd *cobra.Command, details versionDetails) error {
 		if cmdVersion, ok := curr.Annotations["version"]; ok && versions.LessThan(details.CurrentVersion(), cmdVersion) {
 			return fmt.Errorf("%s requires API version %s, but the Docker daemon API version is %s", cmd.CommandPath(), cmdVersion, details.CurrentVersion())
 		}
-		if ost, ok := curr.Annotations["ostype"]; ok && ost != details.ServerInfo().OSType {
-			return fmt.Errorf("%s is only supported on a Docker daemon running on %s, but the Docker daemon is running on %s", cmd.CommandPath(), ost, details.ServerInfo().OSType)
+		si := details.ServerInfo()
+		if ost, ok := curr.Annotations["ostype"]; ok && si.OSType != "" && ost != si.OSType {
```

Comment:
In cases where we were not able to connect with the daemon, we set `Experimental` to "true", but as we don't know the daemon's `OSType`, we can't set that value, so in that case we should ignore these; https://github.com/docker/cli/blob/7240f708087dbea6120477ceb2dcf766d4407804/cli/command/cli.go#L342-L343


We can probably optimise and detect that we didn't have a valid ping, and skip all of these checks, but leaving that for a follow-up.